### PR TITLE
Faculty Component

### PR DESCRIPTION
**Pull request by yurakhomitsky**
Faculty Component has been added.
Component include interface that interacts with server.

### CODE REVIEW

File: src/app/admin/admin-panel-page/admin-panel.component.html [hunk 1]
```diff
@@ -1 +1,3 @@
 <p>Welcome to admin panel!</p>
+<button  mat-flat-button color="primary" (click)="ToFaculty()" >Факультети</button>
```
Comment (LiubomyrTs) [inline src/app/admin/admin-panel-page/admin-panel.component.html]: replace a button with a link routerLink="admin/faculty",

File: src/app/admin/admin-panel-page/admin-panel.component.ts [hunk 2]
```diff
@@ -7,9 +8,11 @@ import { Component, OnInit } from '@angular/core';
 })
 export class AdminPanelComponent implements OnInit {
 
-  constructor() { }
+  constructor(private router: Router) { }
 
   ngOnInit() {
   }
-
+  ToFaculty() {
```
Comment (LiubomyrTs) [inline src/app/admin/admin-panel-page/admin-panel.component.ts]: this method is not needed due to changes in admin-panel component

File: src/app/admin/admin.module.ts [hunk 3]
```diff
@@ -2,16 +2,26 @@ import { NgModule } from '@angular/core';
...
   { path: '', component: AdminPanelComponent },
+  { path: 'faculty', component: FacultyComponent },
 ];
 @NgModule({
-  declarations: [AdminPanelComponent],
+  declarations: [AdminPanelComponent, FacultyComponent],
   imports: [
     CommonModule,
+    FormsModule,
+    FormsModule,
```
Comment (LiubomyrTs) [inline src/app/admin/admin.module.ts]: why 2 times FormsModule ?

File: src/app/admin/faculty/faculty.component.html [hunk 4]
```diff
@@ -0,0 +1,122 @@
...
+        >
+          Додати
+        </button>
+      </div>
+    </form>
+  </div>
+
+  <table
+    mat-table
+    [dataSource]="Facultys"
```
Comment (LiubomyrTs) [inline src/app/admin/faculty/faculty.component.html]: Facultys => Faculties(just a grammar error)

File: src/app/admin/faculty/faculty.component.ts [hunk 5]
```diff
@@ -0,0 +1,101 @@
...
+
+  openDialogWithTemplateRef(templateRef: TemplateRef<any>) {
+    this.dialog.open(templateRef);
+  }
+
+  closeDialog() {
+    this.dialog.closeAll();
+  }
+
+  ngOnInit(): void {
```
Comment (LiubomyrTs) [inline src/app/admin/faculty/faculty.component.ts:62]: Why everything in ngOnInit?

**Review by LiubomyrTs — COMMENTED** [review]
No error handling and loading state visualization. After adding faculty input fields remain in error state.
Also i think we don't  need to show id to admin, I think index would be more appropriate for that column. 
I suggest creating components for new faculty and update faculty.
Layout looks pretty good though.

File: src/app/admin/faculty/faculty.component.scss [hunk 6]
```diff
@@ -0,0 +1,63 @@
...
+  font-size: 1rem;
+  line-height: 1.6;
+}
+.container {
+  max-width: 1000px;
+  margin: 0 auto;
+  padding: 1rem;
+}
+
+.CreateForm {
```
Comment (AndriyIf) [inline src/app/admin/faculty/faculty.component.scss:11]: .create-form. For scss kebab-case is preferable.

File: src/app/admin/faculty/faculty.component.scss [hunk 7]
```diff
@@ -0,0 +1,63 @@
...
+.Form {
+  width: 50%;
+  margin: 20px 0px;
+}
+
+table {
+  width: 100%;
+}
+
+.Form {
```
Comment (AndriyIf) [inline src/app/admin/faculty/faculty.component.scss:45]: .form. Should be combined with the previous .form selector. Class name should be specific, like bla-bla-form

File: src/app/admin/faculty/faculty.component.scss [hunk 8]
```diff
@@ -0,0 +1,63 @@
...
+  margin: 20px auto;
+}
+.example-container {
+  display: flex;
+  flex-direction: column;
+}
+
+.updateForm {
```
Comment (AndriyIf) [inline src/app/admin/faculty/faculty.component.scss:53]: .update-form

File: src/app/admin/faculty/faculty.component.ts [hunk 9]
```diff
@@ -0,0 +1,120 @@
...
+  selector: 'app-faculty',
+  templateUrl: './faculty.component.html',
+  styleUrls: ['./faculty.component.scss']
+})
+export class FacultyComponent implements OnInit {
+
+  loading = false;
+
+  addForm = new FormGroup({
+    faculty_name: new FormControl('',
```
Comment (AndriyIf) [inline src/app/admin/faculty/faculty.component.ts:18]: For js camel case should be used.

File: src/app/admin/faculty/faculty.component.ts [hunk 10]
```diff
@@ -0,0 +1,120 @@
...
+      ]),
+    faculty_description: new FormControl('',
+      [
+        Validators.required,
+        Validators.pattern('[а-яА-ЯіІї ]*')
+      ])
+  });
+
+
+  Faculties: Faculty[] = [];
```
Comment (AndriyIf) [inline src/app/admin/faculty/faculty.component.ts:46]: Capitalized first letter is used only for constructors.

File: src/app/admin/faculty/faculty.component.ts [hunk 11]
```diff
@@ -0,0 +1,120 @@
...
+      .subscribe(response => {
+        this.Faculties = response;
+        this.loading = false;
+      });
+  }
+
+  addFaculty() {
+    this.facultyService.addFaculty({ ...this.addForm.value })
+      .subscribe(response => {
+        this.Faculties.push(response[0]);
```
Comment (AndriyIf) [inline src/app/admin/faculty/faculty.component.ts:78]: Here you can avoid using ViewChild by creating new instance of facilities. this.facilities = [ ...this.facilities, response[0]]

File: src/app/admin/faculty/faculty.component.ts [hunk 12]
```diff
@@ -0,0 +1,120 @@
...
+      faculty_description: element.faculty_description,
+    });
+    this.id = element.faculty_id;
+  }
+
+  updateFaculty() {
+
+    this.facultyService.updateFaculty(this.id, { ...this.updateForm.value })
+      .subscribe(response => {
+        this.getFaculty();
```
Comment (AndriyIf) [inline src/app/admin/faculty/faculty.component.ts:97]: Looks like here is inconsistent approach. Is it possible to change existed data on success?

**Pull request closed without merging**